src/HardwareNFCConnection.cpp

```
@@ -449,6 +449,28 @@ opt_nfc_hal_t* HardwareNFCConnection::getHal() {
     return &hal_;
 }
 
+bool HardwareNFCConnection::getDiagnosticSnapshot(ReaderDiagnostics& out) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)